### ISSUE TITLE
DateTime.Parse throws TypeInitializationException, depends on System.Runtime.InteropServices.PInvoke

### ISSUE THREAD

**Issue by qmfrederik** [dotnet/corefx#10713]
The following code:

```
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Globalization;

namespace DateTimeError
{
    [TestClass]
    public class Class1
    {
        [TestMethod]
        public void Test()
        {
            DateTime.Parse("2015-11-20T10:08:01Z", CultureInfo.CurrentCulture, DateTimeStyles.None);
        }
    }
}
```

results in this error
```
   Test method DateTimeError.Class1.Test threw exception:
System.TypeInitializationException: The type initializer for 'System.Runtime.Versioning.BinaryCompatibility' threw an exception. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime.InteropServices.PInvoke, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
Stack Trace:
    at System.ModuleHandle.ResolveType(RuntimeModule module, Int32 typeToken, IntPtr* typeInstArgs, Int32 typeInstCount, IntPtr* methodInstArgs, Int32 methodInstCount, ObjectHandleOnStack type)
   at System.ModuleHandle.ResolveTypeHandleInternal(RuntimeModule module, Int32 typeToken, RuntimeTypeHandle[] typeInstantiationContext, RuntimeTypeHandle[] methodInstantiationContext)
   at System.Reflection.RuntimeModule.ResolveType(Int32 metadataToken, Type[] genericTypeArguments, Type[] genericMethodArguments)
   at System.Reflection.CustomAttribute.FilterCustomAttributeRecord(CustomAttributeRecord caRecord, MetadataImport scope, Assembly& lastAptcaOkAssembly, RuntimeModule decoratedModule, MetadataToken decoratedToken, RuntimeType attributeFilterType, Boolean mustBeInheritable, Object[] attributes, IList derivedAttributes, RuntimeType& attributeType, IRuntimeMethodInfo& ctor, Boolean& ctorHasParameters, Boolean& isVarArg)
   at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeModule decoratedModule, Int32 decoratedMetadataToken, Int32 pcaCount, RuntimeType attributeFilterType, Boolean mustBeInheritable, IList derivedAttributes, Boolean isDecoratedTargetSecurityTransparent)
   at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeAssembly assembly, RuntimeType caType)
   at System.Attribute.GetCustomAttributes(Assembly element, Type attributeType, Boolean inherit)
   at System.AppDomain.GetTargetFrameworkName()
   at System.Runtime.Versioning.BinaryCompatibility.ReadTargetFrameworkId()
   at System.Runtime.Versioning.BinaryCompatibility.get_AppWasBuiltForFramework()
   at System.Runtime.Versioning.BinaryCompatibility.BinaryCompatibilityMap..ctor()
   at System.Runtime.Versioning.BinaryCompatibility..cctor()
--- End of inner exception stack trace ---
    at System.Globalization.DateTimeFormatInfo.InsertHash(TokenHashValue[] hashTable, String str, TokenType tokenType, Int32 tokenValue)
   at System.Globalization.DateTimeFormatInfo.AddMonthNames(TokenHashValue[] temp, String monthPostfix)
   at System.Globalization.DateTimeFormatInfo.CreateTokenHashTable()
   at System.Globalization.DateTimeFormatInfo.Tokenize(TokenType TokenMask, TokenType& tokenType, Int32& tokenValue, __DTString& str)
   at System.__DTString.GetSeparatorToken(DateTimeFormatInfo dtfi, Int32& indexBeforeSeparator, Char& charBeforeSeparator)
   at System.DateTimeParse.Lex(DS dps, __DTString& str, DateTimeToken& dtok, DateTimeRawInfo& raw, DateTimeResult& result, DateTimeFormatInfo& dtfi, DateTimeStyles styles)
   at System.DateTimeParse.TryParse(String s, DateTimeFormatInfo dtfi, DateTimeStyles styles, DateTimeResult& result)
   at System.DateTimeParse.Parse(String s, DateTimeFormatInfo dtfi, DateTimeStyles styles)
   at DateTimeError.Class1.Test() in C:\Users\FrederikCarlier\Documents\Visual Studio 2015\Projects\DateTimeError\Class1.cs:line 22
```

when running in the context of mstest.

The attached .zip file includes the full source code.

You can find sporadic mentions of other hitting this issue, e.g. https://connect.microsoft.com/VisualStudio/feedback/details/2952488/unable-to-call-datetime-parse-from-within-a-net-core-app-1-0-test-project . You also hit this issue when deserializing JSON that contains a DateTime value.

[DateTimeError.zip](https://github.com/dotnet/corefx/files/413418/DateTimeError.zip)

**Comment by svick** [comment]
`System.Runtime.InteropServices.PInvoke` existed in RC2 version of .Net Core, but it does not exist in the final 1.0 version (see https://github.com/dotnet/corefx/issues/7754). This means the issue is that you're using some old package, which conflicts with using .Net Core 1.0 RTM.

Specifically, it's [`dotnet-test-mstest` 1.0.1-preview](https://www.nuget.org/packages/dotnet-test-mstest/1.0.1-preview), which depends on RC2 version of `Microsoft.NETCore.App`.

There doesn't seem to be a newer version of that package, so, unfortunately, I think you won't be able to use MSTest with .Net Core 1.0 until that package is updated.

**Comment by tarekgh** [comment]
@tijoytom this is not really a DateTime parse issue more than interop packaging issue. could you please follow up with this one> look at @svick comment.

**Comment by tijoytom** [comment]
@yizhang82  This seems to be a fall out from the decision to roll back the new 'System.Runtime.InteropServices.PInvoke' from RTM.

**Comment by yizhang82** [comment]
Yes. Whatever package that uses the stale System.Runtime.InteropServices.PInvoke contract needs to be recompiled to build against the latest set of contracts. We'll follow up with the nuget owner offline to see if they can get the package updated to RTM.

**Comment by AbhitejJohn** [comment]
The MSTest Framework team plan to release an update for dotnet-test-mstest built against the RTM Core Fx by end of this week. Will keep this thread posted.

**Comment by qmfrederik** [comment]
@AbhitejJohn Sounds great, looking forward to the updated `dotnet-test-mstest` package

**Comment by AbhitejJohn** [comment]
@qmfrederik Here is the release of dotnet-test-mstest against RTM Core Fx: https://www.nuget.org/packages/dotnet-test-mstest/1.1.1-preview 
Validated that the scenario above works with this release.

**Comment by qmfrederik** [comment]
@AbhitejJohn The scenario works correctly with the new package, so closing this issue.

Thanks for the quick turnaround on this one!

**Comment by j0nscalet** [comment]
@qmfrederik  @AbhitejJohn verified, fixed. Thanks so much for taking care of this so quickly. You guys are awesome!

**Comment by deppe** [comment]
I'm running into a very similar issue where DateTime.parse is throwing 
`System.TypeInitializationException: The type initializer for 'System.Runtime.Versioning.BinaryCompatibility' threw an exception. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime.InteropServices, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.`

I looked at the project's dependencies and didn't see anything referencing an rtm version of netcoreapp. Are there any tools or ways to identify which of my project's dependencies is pulling in the bad version?

**Comment by tijoytom** [comment]
@deppe Your  exception is about "System.Runtime.InteropServices" , this thread talks about "System.Runtime.InteropServices.PInvoke"

**Comment by deppe** [comment]
Ah, should i open a new issue for this?

**Comment by tijoytom** [comment]
@deppe I dont think it's an issue with the framework , you need to check the version of System.Runtime.InteropServices your app is referencing...

**Comment by deppe** [comment]
@tijoytom What is the best way of figuring that out? The csproj in my project references System.Runtime.InteropServices 4.3.0 which is the most recent version. All references in the .deps.json file are also 4.3.0.  I do, however, see some packages that include 4.1.0 in the project.assets.json file, but I'm not sure if that is relevant/significant.
